### ISSUE TITLE
Sleeping player count keeps getting higher on entering and exiting bed

### ISSUE THREAD

**Issue by poikilos** [minetest/minetest_game#2111]
Player can get messages such as "88/6 players in bed." If someone gets in bed then gets out, the number never goes down, and when the count reaches total player count (such as 6 in this case), the player can optionally skip night (if they opt to not skip, they can keep making number higher), even if only one player is doing this. My server was updated to latest git version less than 2 weeks ago, but the clients have a git version from about 5-7 weeks ago.

**Comment by SmallJoker** [comment]
> If someone gets in bed then gets out, the number never goes down

The number decreases as soon the player stands up (manually or "kicked" after night skip:
https://github.com/minetest/minetest_game/blob/bbb88e6387201ffa6d3231fe9ed9d1b619ace7a9/mods/beds/functions.lua#L64

However, this variable is modified after each change, but the number of entries in `beds.player` isn't counted yet, which would reduce such out-of-sync problems.

**Comment by bell07** [comment]
Same issue in homedecor beds solved by recounting the players each request: https://github.com/minetest-mods/homedecor_modpack/commit/a4ca53cc051837201667f598fefc3aafad59f8bf
Maybe solution for minetest_game too?

Otherwise I thing the out-of-sync appears if the lay_down is called multiple times without standing up, so the 

```
beds.player[name] = 1
beds.pos[name] = pos
player_in_bed = player_in_bed + 1
```
should be

```
if beds.player[name] == nil then
	beds.player[name] = 1
	player_in_bed = player_in_bed + 1
end
beds.pos[name] = pos
```

**Comment by poikilos** [comment]
The code above seems to be implemented at <https://github.com/minetest/minetest_game/blob/master/mods/beds/functions.lua#L62> but the bug still occurs on server & client both at git 3eb363f813bfc4fbca250579848532ef296b3077 (May 16)

**Comment by bell07** [comment]
You refers to the counting players down. I meant the other part, player should not be count up if already in bed for any reason. I prepared a PR for the issue

**Comment by poikilos** [comment]
This issue does not happen in normal use. As I mentioned on the PR, I discovered that the player was using a homedecor bed, and that was fixed when homedecor was updated (homedecor does not use the beds mod). I cannot reproduce the issue I reported, and do not consider it valid anymore--I'm not sure whether a hacked client could reproduce the issue, because I do not fully understand <https://github.com/minetest/minetest_game/issues/1964> and do not know whether that issue has anything to do with the player_in_bed count.
